### PR TITLE
VZ-7380: Update API _types.go files for auto generation of API docs

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request is the first set of changes to update API _types.go files for auto generation of API docs.  The changes are for the following APIs: ingressTrait, verrazzano v1alpah1, and verrazzano v1beta1.  The changes sync-up with what is documented in the public API docs.

### CODE REVIEW

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go [hunk 1]
```diff
@@ -848,22 +956,38 @@ func init() {
 	SchemeBuilder.Register(&Verrazzano{}, &VerrazzanoList{})
 }
 
-// OCI Logging configuration for Fluentd DaemonSet
+// OciLoggingConfiguration is the OCI logging configuration for Fluentd.
```
Comment (rosemarymarano) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:959]: is the OCI logging configuration -> is the Oracle Cloud Infrastructure logging configuration

File: platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ const (
 // +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".status.version",description="The current version of the Verrazzano installation."
 // +genclient
 
-// Verrazzano is the Schema for the verrazzanos API
+// Verrazzano specifies the verrazzanos API.
```
Comment (rosemarymarano) [inline platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go:42]: verrazzanos API -> Verrazzano API

File: platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go [hunk 3]
```diff
@@ -50,125 +50,131 @@ type Verrazzano struct {
 
 // +kubebuilder:object:root=true
 
-// VerrazzanoList contains a list of Verrazzano
+// VerrazzanoList contains a list of Verrazzano.
 type VerrazzanoList struct {
```
Comment (rosemarymarano) [inline platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go:54]: contains a list of Verrazzano -> ???

**Review by rosemarymarano — COMMENTED** [review]
Some edits, not all. Will discuss with you a better strategy for editing these docs.

File: application-operator/apis/oam/v1alpha1/ingresstrait_types.go [hunk 4]
```diff
@@ -14,10 +14,11 @@ import (
 
 // IngressTraitSpec specifies the desired state of an ingress trait.
 type IngressTraitSpec struct {
-	// Rules specifies a list of ingress rules to for an ingress trait.
+	// A list of ingress rules to for an ingress trait.
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/oam/v1alpha1/ingresstrait_types.go:17]: fixed

File: application-operator/apis/oam/v1alpha1/ingresstrait_types.go [hunk 5]
```diff
@@ -31,61 +32,99 @@ type IngressTraitSpec struct {
...
 type IngressPath struct {
-	Path     string               `json:"path,omitempty"`
-	PathType string               `json:"pathType,omitempty"`
-	Policy   *AuthorizationPolicy `json:"authorizationPolicy,omitempty"`
+	// If no path is provided, it defaults to forward slash (/).
+	// +optional
+	Path string `json:"path,omitempty"`
+	// Path type values are case-sensitive and formatted as follows:
+	// <ul><li>`exact`: exact string match</li><li>`prefix`: prefix-based match</li><li>`regex`: regex-based match</li></ul>
+	// If the provided ingress path doesn't contain a `pathType`, it defaults to `prefix` if the path is `/` and `exact`
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/oam/v1alpha1/ingresstrait_types.go:63]: reworded

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go (hunk 1 above)
Comment (bobdonat) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:1083]: fixed

File: platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go (hunk 2 above)
Comment (bobdonat) [reply platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go:42]: fixed

**Review by rosemarymarano — APPROVED** [review]

**Pull request merged**